task/caboose-reader/src/main.rs

```diff
@@ -72,33 +46,19 @@ impl idl::InOrderCabooseImpl for ServerImpl {
         name: [u8; 4],
         data: Leased<W, [u8]>,
     ) -> Result<u32, RequestError<CabooseError>> {
-        if self.pos.is_empty() {
-            return Err(CabooseError::MissingCaboose.into());
+        let reader = self
+            .caboose
```

Comment:
Tiny nit - could `self` hold a `CabooseReader` instead? (Looks like `CabooseReader::new()` is a no-op so this doesn't really matter.)